---
title: "Issues handling didcomm message errors"
date: 2024-02-28 12:47:17 +0000
author: amagyar-iohk
excerpt: >
  Outline of template:
  
  - Title (part of issue title)
  - Submitter (part of issue)
       - Others (optional, if more than one person or an organization)
  - Who (As a...)
      - Details (optional)
  - What (I need...)
      - Details (optional)
  - Why (so I can...)
      - Motivation
           - Use Case or Use Case Category (at least one)
                - allow for proposing a new Use Case Category or linking to a new Use Case proposal issue
      - Details (optional)
categories: decentralized-identity
tags: did-jwt
comments_file: did-jwt-issue-308_comments
permalink: /did-jwt/308/
url: https://github.com/decentralized-identity/did-jwt/issues/308
last_modified_at: 2024-11-20 12:34:20 +0000
---


**URL:** https://github.com/decentralized-identity/did-jwt/issues/308

### Is this a regression?

Yes

### Description

## Brief

When didcomm message returns a `http status code` other than `ok` the message becomes undefined due a try-catch block. This issue's goal is to enable an error management for didcomm errors.
Spec: https://identity.foundation/didcomm-messaging/spec/#problem-reports

FetchApi.ts
```typescript
  async request<T>(
    method: HttpMethod,
    urlStr: string,
    urlParameters: Map<string, string> = new Map(),
    httpHeaders: Map<string, string> = new Map(),
    body?: string | Record<string, any>
  ): Promise<ApiResponse<T>> {
    // ...
    if (response.ok) {
      return new ApiResponse(
        data,
        response.status,
        response.statusText,
        response.headers
      );
    }
    throw new ApiError(response.status, response.statusText, data);
}
```

Mercury.ts
```typescript
  async sendMessageParseMessage(
    message: Domain.Message
  ): Promise<Domain.Message | undefined> {
    const responseBody = await this.sendMessage<any>(message);
    try {
      const responseJSON = JSON.stringify(responseBody);
      return await this.unpackMessage(responseJSON);
    } catch (err) {
      return undefined
    }
  }
```

Any `ApiError` thrown will cause message to return `undefined` as per described in the `Mercury.ts#sendMessageParseMessage` method

### Please provide the exception or error you saw

_No response_

### Please provide the environment you discovered this bug in

_No response_

### Anything else?

_No response_